algoflex_backend/src/gateways/compilation/build.gateway.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  OnGatewayConnection,
+  OnGatewayDisconnect,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { Logger } from '@nestjs/common';
+import { Server } from 'ws';
+import BuildListener from './build_listener';
+
+interface compileRequestEvent {
+  code: string;
+  execute: boolean;
+}
+
+@WebSocketGateway()
+export class BuildGateway implements OnGatewayDisconnect, OnGatewayConnection {
+  @WebSocketServer()
+  server: Server;
+  private buildListener: BuildListener | null = null;
+  private logger: Logger = new Logger('BuildGateway');
+  private execute = false;
+
+  @SubscribeMessage('compile-request')
+  async handleCompileRequest(client: any, event: compileRequestEvent): Promise<void> {
+    if (this.buildListener === null) {
+      this.buildListener = await BuildListener.create(
+        event.code,
+        `
+      {
+        "exercice": [
+            {
+                "entree": ["30"],
+                "sorties": ["a"],
+                "maxtime": 30000,
+                "maxmemory": 10
+            },
+            {
+                "entree": ["8"],
+                "sorties": ["a"],
+                "maxtime": 30000,
+                "maxmemory": 10
+            }
+        ]
+    }`
+      ); // TODO : a changer
+      this.execute = event.execute;
+
+      client.send(
+        JSON.stringify({
+          state: 1,
+          executeLink: this.buildListener.getExecuteLink(),
+          compileLink: this.buildListener.getCompileLink(),
+        })
+      );
+    }
+  }
+
+  @SubscribeMessage('execute-request')
+  async handle(client: any): Promise<void> {
+    if (this.buildListener !== null) {
```

Comment:
```suggestion
    if (this.buildListener === null) {
      return
    }
```
Ici perso j'inverserai peut-être la condition pour pouvoir sortir le reste de la logique du if